demo/pyproject.toml

```diff
@@ -83,33 +83,17 @@ author_email = "tiberius@beeware.org"
 formal_name = "Toga Demo"
 description = "A demonstration of the capabilities of the Toga widget toolkit."
 sources = ["toga_demo"]
-requires = ["../core"]
 
```

Comment:
toga-demo lists `toga` as a dynamic dependency, so we don't need explicit references to each backend and core.